lib/packagekit-glib2/pk-offline.c

```diff
@@ -473,17 +474,18 @@ pk_offline_get_prepared_ids (GError **error)
 	}
 
 	keyfile = g_key_file_new ();
+	g_key_file_set_list_separator (keyfile, ',');
 	if (!g_key_file_load_from_data (keyfile, data, -1, G_KEY_FILE_NONE, &error_local)) {
 		/* fall back to previous plain text file format for backwards compatibility */
 		return g_strsplit (data, "\n", -1);
 	}
 
-	prepared_ids = g_key_file_get_string (keyfile, "update", "prepared_ids", error);
-	if (prepared_ids == NULL)
+	prepared_ids = g_key_file_get_string_list (keyfile, "update", "prepared_ids", &prepared_ids_size, error);
+
+	if (prepared_ids == NULL || prepared_ids_size == 0)
 		return NULL;
```

Comment:
in case the `prepared_ids` is empty, but non-NULL, the pointer is leaked here.

Reply:
Of course, that's my fault. Will do a bit later today.